scripts/download.pl

```diff
@@ -70,27 +70,50 @@ ()
 	return undef;
 }
 
-sub download_cmd($) {
+sub download_cmd {
 	my $url = shift;
 	my $have_curl = 0;
-
-	if (open CURL, "curl --version 2>/dev/null |") {
+	my $have_aria2c = 0;
+	my $fn = shift;
+	my @mrs = @_;
+	my $murl = "'$url'";
+
+	my @chArray = ('a'..'z', 'A'..'Z', 0..9);
+	my $rfn = join '', map{ $chArray[int rand @chArray] } 0..9;
+	if (open CURL, '-|', 'curl', '--version') {
 		if (defined(my $line = readline CURL)) {
 			$have_curl = 1 if $line =~ /^curl /;
 		}
 		close CURL;
 	}
+	if (open ARIA2C, '-|', 'aria2c', '--version') {
+		if (defined(my $line = readline ARIA2C)) {
+			$have_aria2c = 1 if $line =~ /^aria2 /;
+		}
+		close ARIA2C;
+	}
+
+	for my $el (@mrs) {
+		$murl = $murl." '$el/$fn'";
+	}
 
-	return $have_curl
-		? (qw(curl -f --connect-timeout 20 --retry 5 --location),
+	if ($have_aria2c) {
+		if ($check_certificate) {
+			return ("aria2c --stderr $murl -c -x2 -s10 -j10 -k1M --server-stat-of=/dev/shm/spp --server-stat-if=/dev/shm/spp -d /dev/shm -o $rfn; cat /dev/shm/$rfn; rm /dev/shm/$rfn");
+		} else {
+			return ("aria2c --stderr $murl -c -x2 -s10 -j10 -k1M --server-stat-of=/dev/shm/spp --server-stat-if=/dev/shm/spp -d /dev/shm -o $rfn --check-certificate=false; cat /dev/shm/$rfn; rm /dev/shm/$rfn");
+		}
```

Comment:
```suggestion
		return join(" ", "touch /dev/shm/${rfn}_spp;",
			qw(aria2c --stderr -c -x2 -s10 -j10 -k1M), $murl,
			$check_certificate ? () : '--check-certificate=false',
			"--server-stat-of=/dev/shm/${rfn}_spp",
			"--server-stat-if=/dev/shm/${rfn}_spp",
			"-d /dev/shm -o $rfn;",
			"cat /dev/shm/$rfn;", "rm /dev/shm/$rfn /dev/shm/${rfn}_spp");
```

What do you think of this solution?

Also I notice that in your implementation you didn't create the ssp in the first place (so it wasn't actually used) and also it's used by all the packages. Is that intended?